### DISCUSSION TITLE
表格中同时使用强制换行和自动换行对不齐

### DISCUSSION THREAD

**Discussion by qiaoxiaoyue** [sjtug/SJTUThesis#782] [提问] [answered]
图片给了个例子，提出者一栏中使用\tabincell{c}强制换行并居中，后面优势特点和劣势不足两栏通过设置列宽令其自动换行，出现图中所示的上下无法对齐。
![截屏2022-04-09 下午3 00 16](https://user-images.githubusercontent.com/28865025/162560810-2c597b49-2a2d-4b18-905d-7102d00215aa.png)
对后面两栏使用左对齐强制换行后上下能对齐，但不好把握文字到表格线的距离，所以还是想要同时使用强制换行和自动换行。

**Comment by LogCreative** [comment, marked as answer]
`\tabincell` 应该是自定义的宏，或许你参考了 [这个博客](https://www.cnblogs.com/tsingke/p/5506354.html)，那么你应当修改宏的定义为
```latex
\newcommand{\tabincell}[2]{\begin{tabular}[t]{@{}#1@{}}#2\end{tabular}}
\begin{document} % 添加 [t] 参数
```
更多信息可以参考 [learnlatex.org 更多：表格](https://github.com/CTeX-org/learnlatex.github.io/blob/zh-Hans/zh-Hans/more-08.md#%E7%BA%B5%E5%90%91%E6%8A%80%E5%B7%A7)。

<img width="312" alt="image" src="https://user-images.githubusercontent.com/61653082/162562393-89855962-9f9d-4bf6-a22f-27d05742af55.png">

<details>
<summary>MWE</summary>

```latex
\documentclass{ctexart}
\usepackage{array}
\newcommand{\tabincell}[2]{\begin{tabular}[t]{@{}#1@{}}#2\end{tabular}}
\begin{document}
    \begin{table}
        \begin{tabular}{|l|l|l|p{3cm}|p{3cm}|}
            \hline
            \multicolumn{1}{|c|}{时间} & \multicolumn{1}{c|}{提出者} & \multicolumn{1}{c|}{原理} & \multicolumn{1}{c|}{优势特点} & \multicolumn{1}{c|}{劣势不足} \\
            \hline
            1985 & \tabincell{c}{Raugh M R \\ 斯坦福大学} & 群论 & 确定自校准基本模型，奠定理论基础 & 未实验论证，对噪声敏感 \\
            \hline
        \end{tabular}
    \end{table}
\end{document}
```

</details>

**Reply by qiaoxiaoyue** [reply]
十分感谢，回答的很认真详细。

**Comment by AlexaraWu** [comment]
直接用 `makecell` 包不香吗？

![image](https://user-images.githubusercontent.com/15205102/162573569-fce14502-284a-4bed-b7e6-604e27625548.png)

<details>
<summary>MWE</summary>

```latex
\documentclass{ctexart}
\usepackage{array}
\usepackage{makecell}
\renewcommand\theadfont{\normalsize}
\begin{document}
\begin{table}
  \begin{tabular}{|l|l|l|m{3cm}|m{3cm}|}
    \hline
    \thead{时间} & \thead{提出者} & \thead{原理} & \thead{优势特点} & \thead{劣势不足} \\
    \hline
    1985 & \makecell{Raugh M R\\ 斯坦福大学} & 群论 & 确定自校准基本模型，奠定理论基础 & 未实验论证，对噪声敏感 \\
    \hline
  \end{tabular}
\end{table}
\end{document}
```

</details>

**Reply by LogCreative** [reply]
确实方便👍
